sound/soc/sof/sof-audio.c

```diff
@@ -232,6 +260,7 @@ int sof_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget *swidget)
 
 use_count_dec:
 	swidget->use_count--;
+	snd_sof_dsp_core_put(sdev, core);
```

Comment:
is the order important here? If you could do
```
core_put:
	snd_sof_dsp_core_put(sdev, core);
use_count_dec:
	swidget->use_count--;
```
you could still use `goto use_count_dec` in the first error case